policy-engine/src/graph.rs

```diff
@@ -189,4 +240,185 @@ mod tests {
             graph::GraphError::MissingParams(vec!["id".to_string()])
         );
     }
+
+    #[test]
+    fn failed_plugin_execution() -> Result<(), Box<Error>> {
+        use std::str::FromStr;
+
+        let mut rt = common_init();
+        let mandatory_params = vec!["channel".to_string()].into_iter().collect();
+        let state = AppState {
+            mandatory_params,
+            upstream: hyper::Uri::from_str(&mockito::server_url())?,
+            ..Default::default()
+        };
+
+        let http_req = actix_web::test::TestRequest::get()
+            .data(state)
+            .uri(&format!("{}?channel=':'", "http://doesn.really/matter"))
+            .header(
+                http::header::ACCEPT,
+                http::header::HeaderValue::from_static(cincinnati::CONTENT_TYPE),
+            )
+            .to_http_request();
+
+        let graph_call = graph::index(http_req);
+
+        let _m = mockito::mock("GET", "/")
+            .with_status(200)
+            .with_header("content-type", "application/json")
+            .with_body(r#"{"nodes":[],"edges":[]}"#)
+            .create();
+
+        match rt.block_on(graph_call) {
+            Err(graph::GraphError::FailedPluginExecution(ref msg))
+                if msg.contains("does not match regex") =>
+            {
+                Ok(())
+            }
+            res => Err(format!("expected FailedPluginExecution error, got: {:?}", res).into()),
+        }
+    }
+
+    #[test]
+    fn httpserver_must_respond_with_appropriate_json() -> Result<(), Box<Error>> {
```

Comment:
I think this is only testing JSON error responses. Something like `webservice_json_error_reponses` would be a bit more descriptive here.